### PR TITLE
update the files for transfer exit with a new method

### PR DESCRIPTION
**Pull request by lodug**
Added the new method requestTRansferExitRequest that will facilitate the transfer exit called in handle frames.

### CODE REVIEW

File: src/ota/request_transfer_exit/src/RequestTransferExit.cpp [hunk 1]
```diff
@@ -45,3 +50,57 @@ bool RequestTransferExit::requestTransferExit(int id, bool transferSuccess)
...
+    /* Check the frame data */
+    if (request_transfer_exit_data.size() < 3)
+    {
+        /* Incorrect message length or invalid format - prepare a negative response */
+        response.push_back(0x03); /* PCI */
+        response.push_back(0x7F); /* Negative response */
+        response.push_back(0x37); /* Service ID */
+        response.push_back(0x13); /* Incorrect message length or invalid format */
+        /* Send the negative response frame */ 
+        generate_frames.sendFrame(can_id, response);
```
Comment (RubenNatanael) [inline src/ota/request_transfer_exit/src/RequestTransferExit.cpp:68]: You can use the negativeResponse method from generateFrames to create a NRC, it's not necessary to create the frame from scratch.
Same in other parts of the code

File: src/ota/request_transfer_exit/include/RequestTransferExit.h [hunk 2]
```diff
@@ -15,18 +15,26 @@
 #include <functional>
 
 #include "../../utils/include/Logger.h"
+#include "../../../utils/include/GenerateFrames.h"
+#include "../../utils/include/MemoryManager.h"
```
Comment (RubenNatanael) [inline src/ota/request_transfer_exit/include/RequestTransferExit.h:19]: Do you use this library?

File: src/ota/request_transfer_exit/src/RequestTransferExit.cpp (hunk 1 above)
Comment (lodug) [reply src/ota/request_transfer_exit/src/RequestTransferExit.cpp]: done

File: src/ota/request_transfer_exit/include/RequestTransferExit.h (hunk 2 above)
Comment (lodug) [reply src/ota/request_transfer_exit/include/RequestTransferExit.h]: no, I deleted it

File: src/ota/request_transfer_exit/include/RequestTransferExit.h [hunk 3]
```diff
@@ -15,18 +15,25 @@
...
 {
-public:
+public:    
     /**
    * @brief Constructor.   
    */
-    /* constructor to initialize the transfer exit */
-    RequestTransferExit(Logger& RTESLogger);
+    // /* constructor to initialize the transfer exit */
+    // RequestTransferExit(Logger& RTESLogger);
```
Comment (Miruna-Maria) [inline src/ota/request_transfer_exit/include/RequestTransferExit.h:30]: if these are not needed you may consider to remove them, or keep them with /**/

Comment (lodug) [reply src/ota/request_transfer_exit/include/RequestTransferExit.h]: done, removed it

**Review by Miruna-Maria — APPROVED** [review]

**Review by Mihnea9901 — APPROVED** [review]

**Pull request merged**